src/mongo/db/storage/wiredtiger/wiredtiger_backup_cursor_hooks.cpp

```diff
@@ -194,6 +194,17 @@ BackupCursorState WiredTigerBackupCursorHooks::openBackupCursor(
         builder << "checkpointTimestamp" << checkpointTimestamp.get();
     }
 
+    // Copy backup options to metadata
+    builder << "disableIncrementalBackup" << options.disableIncrementalBackup;
+    builder << "incrementalBackup" << options.incrementalBackup;
+    builder << "blockSize" << options.blockSizeMB;
+    if (options.thisBackupName) {
+        builder << "thisBackupName" << *options.thisBackupName;
+    }
+    if (options.srcBackupName) {
+        builder << "srcBackupName" << *options.srcBackupName;
+    }
+
```

Comment:
I'd convert this block of code into a member function of the `StorageEngine::BackupOptions` class. For instance,
```
void StorageEngine::BackupOptions::serialize(BSONObjBuilder* b) const {
    <the_suggested_code_block_is_here>
}
```
Besides making the calling code shorter and clearer, it also enables code reuse. As a bonus, it would be easier to log an object of `BackupOptions` as a LOGV2 attribute.

By the way, complementary [block](https://github.com/percona/percona-server-mongodb/blob/master/src/mongo/db/pipeline/document_source_backup_cursor.cpp#L136) seems to be a good candidate for transforming into the body of the `BackupOptions(BSONObj& o)` constructor.